source-postgres/prerequisites.go

```diff
@@ -22,31 +26,107 @@ func (db *postgresDatabase) SetupPrerequisites(ctx context.Context) []error {
 }
 
 func (db *postgresDatabase) prerequisiteLogicalReplication(ctx context.Context) error {
-	// TODO: Verify that 'wal_level = LOGICAL'
+	var level string
+	if err := db.conn.QueryRow(ctx, `SHOW wal_level;`).Scan(&level); err != nil {
+		return fmt.Errorf("unable to query 'wal_level' system variable: %w", err)
+	} else if level != "logical" {
+		return fmt.Errorf("logical replication isn't enabled: current wal_level = %q", level)
+	}
 	return nil
 }
 
 func (db *postgresDatabase) prerequisiteReplicationUser(ctx context.Context) error {
-	// TODO: Verify that the capture user has REPLICATION permission
+	var query = fmt.Sprintf(`SELECT rolreplication FROM pg_catalog.pg_roles WHERE rolname = '%s'`, db.config.User)
+	var replication bool
+	if err := db.conn.QueryRow(ctx, query).Scan(&replication); err != nil {
+		return fmt.Errorf("error querying REPLICATION role for user %q: %w", db.config.User, err)
+	}
+	if !replication {
+		return fmt.Errorf("user %q must have the REPLICATION role", db.config.User)
+	}
 	return nil
 }
 
 func (db *postgresDatabase) prerequisiteReplicationSlot(ctx context.Context) error {
-	// TODO: Verify that the replication slot exists
+	// If the replication slot already exists then we're satisfied
+	var slotName = db.config.Advanced.SlotName
+	var logEntry = logrus.WithField("slot", slotName)
+	var count int
+	if err := db.conn.QueryRow(ctx, fmt.Sprintf(`SELECT COUNT(*) FROM pg_catalog.pg_replication_slots WHERE slot_name = '%s' AND slot_type = 'logical';`, slotName)).Scan(&count); err != nil {
+		return fmt.Errorf("error querying replication slots: %w", err)
+	}
+	if count == 1 {
```

Comment:
I'm assuming its not possible for this and other similar checks to be > 1?

Reply:
That's correct, the checks where I've written `== 1` are querying by unique columns so it should be impossible for the results to be > 1.

And if things are somehow so badly broken that we get _multiple_ results for a particular replication slot name then it's probably better to fail anyway.